src/packed-object-index.ts

```diff
@@ -1,15 +1,33 @@
-import { join as pathJoin, relative as pathRelative } from 'path';
+import { join as pathJoin, relative as pathRelative, basename } from 'path';
 import { promises as fsPromises } from 'fs';
 import { PackContent, readPackFile } from './packed-pack.js';
 import {
     InternalGitObjectContent,
     InternalGitObjectHeader,
     PackedObjectType,
-    ObjectsTypeStat
+    ObjectsTypeStat,
+    CruftPackMode
 } from './types.js';
 import { objectsStatFromTypes, sumObjectsStat } from './utils/stat.js';
 
-export async function createPackedObjectIndex(gitdir: string) {
+interface PackedObjectIndexOptions {
+    /**
+     * absolute path to .git dir
+     */
+    gitdir: string;
+    /**
+     * Controls the inclusion of cruft packs in packed objects procession.
+     */
+    cruftPacks: CruftPackMode;
+}
+
+/**
+ * Creates an index object that provides information about git objects stored within pack files
+ * @param options
+ */
+export async function createPackedObjectIndex(options: PackedObjectIndexOptions) {
```

Comment:
If you change the signature to the following, you will avoid unnecessary type declarations (less code):
```suggestion
export async function createPackedObjectIndex(gitdir: string, options: GitReaderOptions) {
```

Reply:
yup, I considered this, however `PackedObjectIndexOptions` and `GitReaderOptions` are somewhat different. The first one is a subset of the later one.

Reply:
There is no need for a specialized options for internal parts. It's also simplify things and number of changes when a new options will be added